### PR TITLE
Improve donate aa1

### PR DESCRIPTION
**Pull request by Aichnerc**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [x] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently when logged in, the message after donating can not be set.
- Currently there is no hash generated for each transaction.
- Currently it is not possible to sign up without errors.
- Currently the version is not being displayed in the footer.

**What is the new behavior (if this is a feature change)?**
- The message can now be set when donating with a logged in user.
- A hash is being created for each transaction. This is being used to display the verified donation badge.
- The sign up has finally been fixed.
- The version along with other information is now being displayed in the footer.

**Other information**:
- Patch (1.4.1)

### CODE REVIEW

File: src/components/pages/DonatePage/index.jsx [hunk 1]
```diff
@@ -199,11 +202,14 @@ class DonatePage extends React.Component {
...
-                                this.state.donationmessage
+                              this.setState({ savedMessage: true }, () =>
+                                this.props.writeDonationMessage(
+                                  selectedDonation,
+                                  this.state.donationmessage
+                                )
                               );
-                              this.setState({ savedMessage: true });
+                            } else {
+                              console.log("bruh");
```
Comment (LucaAllmaier) [inline src/components/pages/DonatePage/index.jsx:212]: is this intentional? xd

Comment (Aichnerc) [reply src/components/pages/DonatePage/index.jsx:212]: People who forget to remove strange console logs 🤡

File: src/components/molecules/Footer/index.jsx [hunk 2]
```diff
@@ -114,7 +114,55 @@ class Footer extends React.Component {
             </p>
           </MDBContainer>
           <MDBContainer fluid>
-            &copy; {new Date().getFullYear()} Copyright: SithCult
+            <div>
+              &copy; {new Date().getFullYear()} Copyright: SithCult
```
Comment (kleberbaum) [inline src/components/molecules/Footer/index.jsx:118]:
```suggestion
              &copy; 2020-{new Date().getFullYear()} Copyright: SithCult
```

Comment (Aichnerc) [reply src/components/molecules/Footer/index.jsx:118]: 2020-2020 might not be a very beautiful solution.

**Pull request merged**